consensus/src/txn_hash_and_authenticator_deduper.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    counters::{TXN_DEDUP_FILTERED, TXN_DEDUP_SECONDS},
+    transaction_deduper::TransactionDeduper,
+};
+use aptos_types::transaction::SignedTransaction;
+use rayon::prelude::*;
+use std::collections::{HashMap, HashSet};
+
+pub struct TxnHashAndAuthenticatorDeduper {}
+
+impl TransactionDeduper for TxnHashAndAuthenticatorDeduper {
+    fn dedup(&self, transactions: Vec<SignedTransaction>) -> Vec<SignedTransaction> {
+        let _timer = TXN_DEDUP_SECONDS.start_timer();
+        let mut seen = HashMap::new();
+        let mut is_possible_duplicate = false;
+        let mut possible_duplicates = vec![false; transactions.len()];
+        for (i, txn) in transactions.iter().enumerate() {
+            match seen.get(&(txn.sender(), txn.sequence_number())) {
```

Comment:
should be a bit better with matching seen.entry() if the rust compiler can be made happy.